tests/kafkatest/services/kafka/kafka.py

```diff
@@ -143,6 +145,7 @@ def security_config(self):
 
     def open_port(self, protocol):
         self.port_mappings[protocol] = self.port_mappings[protocol]._replace(open=True)
+        self.broker_ports.append(self.port_mappings[protocol].number)
```

Comment:
Is `open_port` guaranteed to be called once? If not, `broker_ports` should be a `set` instead of a `list`. Alternatively, it could be an accessor that gets the full *current* list from `port_mappings`.